### PR TITLE
Feature/javascript

### PR DESCRIPTION
**Pull request by xavierang**

### CODE REVIEW

File: src/html/shared/signup.html [hunk 1]
```diff
@@ -4,12 +4,15 @@
...
                 <h3 class="c-signup__heading">Sign me up for more information</h3>
-                <form action="submit" class="o-form">
+                <form action="https://github.us17.list-manage.com/subscribe/post" method="POST" class="o-form" data-module="Form">
+                    <input type="hidden" name="u" value="43373efd1b02bff2a76a8ce96">
+                    <input type="hidden" name="id" value="58ed49b77c">
+
                     <label class="u-hidden-visually">Sign up with email address</label>
-                    <input type="email" placeholder="Your email address" class="o-form__input" required>
-                    <button class="c-btn  c-btn--secondary  c-btn--small o-form__submit">Sign up</button>
+                    <input id="email" name="MERGE0" type="email" placeholder="Your email address" class="o-form__input" data-module="Form" required>
```
Comment (xavianaxw) [inline src/html/shared/signup.html]: don't think data-module="Form" should be used here. Best remove `data-module` and select the element within the `Form` module.

so something like

`this.el.querySelector("#email").addEventListener()`

File: src/javascripts/modules/CollapseAccordion.js [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+  }
+
+  toggleClass(t) {
+    t.classList.toggle("is-collapsed");
+    const sprites = t.children[0].children;
+    for (const sprite of sprites) {
+      sprite.classList.toggle("u-hide");
+    }
+  }
+}
```
Comment (xavianaxw) [inline src/javascripts/modules/CollapseAccordion.js]: Looking at your html

```
<div class="c-accordion__wrapper">
    <p class="c-accordion__header" aria-expanded="false" aria-controls="accordion-3" data-module="CollapseAccordion">
        Footer Section 3 
        {{ sprite("plus", "0 0 1 1", "c-icon c-icon--tiny c-icon--darkbg c-accordion__icon") }}
        {{ sprite("chev", "0 0 1 1", "c-icon c-icon--tiny c-icon--darkbg c-accordion__icon  u-hide") }}
    </p>
    <ul class="o-list-bare  c-accordion__list" id="accordion-3" aria-hidden="true">
        <li class="o-list-bare__item c-accordion__panel_text"><a></a></li>
        <li class="o-list-bare__item c-accordion__panel_text"><a></a></li>
        <li class="o-list-bare__item c-accordion__panel_text"><a></a></li>
    </ul>
</div>
```

Why would you `this.el.parentElement` and then `t.children[0]` ?

You could refactor your code to look like this

```
this.el.addEventListener("click", e => {
    e.preventDefault();
    this.toggleClass(this.el);
});

toggleClass(t) {
    t.parentElement.toggle("is-collapsed"); // selects .c-accordion__wrapper
    
    const sprites = Array.from(t.querySelectorAll("sprite")); // Array.from makes it into an Iterable item
    for (const sprite of sprites) {
        sprite.classList.toggle("u-hide");
    }
}
```

Also making it into a function doesn't have much use here but it's fine

File: src/javascripts/modules/Form.js [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+  constructor(el) {
+    this.el = el;
+
+    this.el.addEventListener("keyup", e => {
+      if (e.target.value !== "") {
+        this.el.classList.add("is-filled");
+      } else {
+        this.el.classList.remove("is-filled");
+      }
+    });
```
Comment (xavianaxw) [inline src/javascripts/modules/Form.js]: This bit of code makes your form have keyup event as well. I'm assuming this are for input fields only?

File: src/html/shared/menu.html [hunk 4]
```diff
@@ -1,16 +1,16 @@
-<div class="c-menu  u-hide@tablet">
+<div class="c-menu  js-menu  u-hide@tablet" id="menu">
```
Comment (xavianaxw) [inline src/html/shared/menu.html]: does `id=menu` gets queried anywhere?

File: src/javascripts/modules/ScrollTo.js [hunk 5]
```diff
@@ -0,0 +1,62 @@
...
+            document
+              .querySelector(".c-video__video")
+              .classList.add("is-playing");
+            document.querySelector(".c-video__video").src += "&amp;autoplay=1";
+            break;
+          default:
+            // do nothing
+            break;
+        }
+      } else if (this.el.localName === "a") {
```
Comment (xavianaxw) [inline src/javascripts/modules/ScrollTo.js]: instead of `localName` i'll use `tagName`

**Review by xavianaxw — CHANGES_REQUESTED** [review]
That's all

File: src/javascripts/modules/Form.js (hunk 3 above)
Comment (xavierang) [reply src/javascripts/modules/Form.js]: Yea only for the input. Removing data-module from input and calling listener on this.el.querySelector("#email") from before should fix it.

Comment (xavianaxw) [reply src/javascripts/modules/Form.js]: 🌮  +1

**Pull request merged**